src/dataconnect/provisionCloudSql.ts

```diff
@@ -73,34 +77,44 @@ export async function provisionCloudSql(args: {
           `\nThis instance is provided under the terms of the Data Connect free trial ${freeTrialTermsLink()}` +
           `\nMonitor the progress at ${cloudSqlAdminClient.instanceConsoleLink(projectId, instanceId)}`,
       );
-    const newInstance = await promiseWithSpinner(
-      () =>
-        cloudSqlAdminClient.createInstance(
-          projectId,
-          locationId,
-          instanceId,
-          enableGoogleMlIntegration,
-          waitForCreation,
-        ),
-      "Creating your instance...",
-    );
-    if (newInstance) {
-      silent || utils.logLabeledBullet("dataconnect", "Instance created");
-      connectionName = newInstance?.connectionName || "";
-    } else {
-      silent ||
-        utils.logLabeledBullet(
-          "dataconnect",
-          "Cloud SQL instance creation started - it should be ready shortly. Database and users will be created on your next deploy.",
-        );
-      return connectionName;
+    if (!dryRun) {
+      const newInstance = await promiseWithSpinner(
+        () =>
+          cloudSqlAdminClient.createInstance(
+            projectId,
+            locationId,
+            instanceId,
+            enableGoogleMlIntegration,
+            waitForCreation,
+          ),
+        "Creating your instance...",
+      );
+      if (newInstance) {
+        silent || utils.logLabeledBullet("dataconnect", "Instance created");
+        connectionName = newInstance?.connectionName || "";
+      } else {
+        silent ||
+          utils.logLabeledBullet(
+            "dataconnect",
+            "Cloud SQL instance creation started - it should be ready shortly. Database and users will be created on your next deploy.",
+          );
+        return connectionName;
+      }
     }
   }
+
   try {
     await cloudSqlAdminClient.getDatabase(projectId, instanceId, databaseId);
     silent || utils.logLabeledBullet("dataconnect", `Found existing database ${databaseId}.`);
   } catch (err: any) {
     if (err.status === 404) {
+      if (dryRun) {
+        silent ||
+          utils.logLabeledBullet(
+            "dataconnect",
+            `Database ${databaseId} not found. It will be created on your next deploy.`,
+          );
+      }
       // Create the database if not found.
```

Comment:
Does this need to be wrapped in an `else`? Otherwise it'll run even in `dryRun`, right?

Reply:
Yes it does, good catch